### ISSUE TITLE
CCAPS (all flavours) - check all factors of rho in code and wiki

### ISSUE THREAD

**Issue by JOThurgood** [JOThurgood/SimpleCFD#115]

**Comment by JOThurgood** [comment]
Currently in the variable density versions of the code the following happens in terms of factors of rho

The old ( "t=n" ) density is used in 

- step 2: this is uncontroversial
- step 3: There is a question over whether advect_dens should occur before this (immediately after step_2) and then a half time density can be computed for use in this step (you would also need to flesh out the code for step 3 which assumes a lot of things don't change but now do with the different factor of rho). This is what seems to happen in `lm_atm` in `pyro2`.

The half time (t=n+1/2) density is used in:
- step 4: gravitational source /forcing for the beta/rho or 1/rho coefficient (seems appropriate). (_NB_ only in the atmospheric code, the other versions have not been updated yet and use the n+1 value for rho).

The new (t=n+1) density is used in:

- step 4: for the 1/rho coefficient in front of the lagged pressure (this seems _inappropriate_)
- all of step 5: seems appropriate.

In other words:

-  Should `advect_dens` be called before `step_3`, and used in the updated interface states ? (as it is in ` pyro2 lm_atm`? Remember, step_3 would have to be fleshed out in this case - it basically assumes nothing changes except for a new up-winding. 

- Should you use a half time rho for all coefficients in step_4? (i.e. not just for gravity in the case of the atmospheric version, and if so update other versions consistently.)
